src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt

```diff
@@ -14,24 +14,40 @@ import io.jsonwebtoken.Jwts
 class AuthenticationFilter : ContainerRequestFilter {
     companion object {
         const val cookieName = "Jwt"
-        const val ignoreThisUrl = "jwt/login"
-    }
+        const val ignoreThisUrl = "api/login"
+        const val employeeIdHeader = "X-Employee-Id"
+        const val companyIdHeader = "X-Company-Id"    }
 
     override fun filter(requestContext: ContainerRequestContext) {
 
         if (requestContext.uriInfo.path == ignoreThisUrl) return
 
         val jwtCookie = requestContext.cookies[cookieName]?.value
-        verify(jwtCookie, requestContext)
+        val claims = verifyAndExtractClaims(jwtCookie, requestContext)
+        claims?.let{
+            requestContext.headers.add(employeeIdHeader, it["employeeId"].toString())
+            requestContext.headers.add(companyIdHeader, it["companyId"].toString())
+        }
```

Comment:
this is one way of doing it, and it's ok. see if there's a way that you can create a User object here and store that as a request attribute...